cmd/publicize/server.go

```diff
@@ -150,37 +151,43 @@ func (s *server) mergeAndPushToRemote(sourceOrg, sourceRepo, destOrg, destRepo s
 	}()
 
 	if err := repoClient.Checkout(branch); err != nil {
-		return fmt.Errorf("couldn't checkout to branch %s: %w", branch, err)
+		return "", fmt.Errorf("couldn't checkout to branch %s: %w", branch, err)
 	}
 
 	if err := repoClient.FetchFromRemote(sourceRemoteResolver, branch); err != nil {
-		return fmt.Errorf("couldn't fetch from the downstream repository: %w", err)
+		return "", fmt.Errorf("couldn't fetch from the downstream repository: %w", err)
 	}
 
 	if err := repoClient.Config("user.name", s.gitName); err != nil {
-		return fmt.Errorf("couldn't set config user.name=%s: %w", s.gitName, err)
+		return "", fmt.Errorf("couldn't set config user.name=%s: %w", s.gitName, err)
 	}
 
 	if err := repoClient.Config("user.email", s.gitEmail); err != nil {
-		return fmt.Errorf("couldn't set config user.name=%s: %w", s.gitEmail, err)
+		return "", fmt.Errorf("couldn't set config user.name=%s: %w", s.gitEmail, err)
 	}
 
 	merged, err := repoClient.MergeWithStrategy("FETCH_HEAD", "merge", git.MergeOpt{CommitMessage: "DPTP reconciliation from downstream"})
 	if err != nil {
-		return fmt.Errorf("couldn't merge %s/%s, merge --abort failed with reason: %w", destOrg, destRepo, err)
+		return "", fmt.Errorf("couldn't merge %s/%s, merge --abort failed with reason: %w", destOrg, destRepo, err)
 	}
 
 	if !merged {
-		return fmt.Errorf("couldn't merge %s/%s, due to merge conflict. You will need to create a new PR in %s/%s which merges/resolves from %s/%s. Once this PR merges, you can then use /publicize there to merge all changes into the the public repository.", destOrg, destRepo, sourceOrg, sourceRepo, destOrg, destRepo)
+		return "", fmt.Errorf("couldn't merge %s/%s, due to merge conflict. You will need to create a new PR in %s/%s which merges/resolves from %s/%s. Once this PR merges, you can then use /publicize there to merge all changes into the the public repository.", destOrg, destRepo, sourceOrg, sourceRepo, destOrg, destRepo)
 	}
 
 	if !dry {
 		if err := repoClient.PushToCentral(branch, false); err != nil {
-			return fmt.Errorf("couldn't push to upstream repository: %w", err)
+			return "", fmt.Errorf("couldn't push to upstream repository: %w", err)
 		}
 	}
 
-	return nil
+	refs, err := repoClient.ShowRef(branch)
```

Comment:
This is unnecessarily broad and that makes the guess below (get first SHA) not as reliable as possible. You can ask  for `refs/head/$branch` ("head of a local branch $branch", filtering out remotes and tags)  and then you can (should) check you got *exactly* a single line, so you know you have a reliable SHA.